File: backend/app/schemas/actions.py
from typing import Literal

from pydantic import BaseModel

DocumentActionType = Literal["replace_selection"]


class ReplaceSelectionAction(BaseModel):
    type: Literal["replace_selection"]
    title: str
    original_text: str
    proposed_text: str
    rationale: str
    created_at: str
    rationale_source: Literal["llm", "fallback"] = "llm"


DocumentAction = ReplaceSelectionAction

File: frontend/src/taskpane/types/actions.ts
export type DocumentActionType = "replace_selection";

export type ReplaceSelectionAction = {
  type: "replace_selection";
  title: string;
  originalText: string;
  proposedText: string;
  rationale: string;
  rationaleSource?: "llm" | "fallback";
  createdAt: string;
};

export type DocumentAction = ReplaceSelectionAction;

export type DocumentActionResult = {
  success: boolean;
  message: string;
};
